sfge/src/main/java/com/salesforce/graph/build/AbstractApexVertexBuilder.java

```diff
@@ -66,11 +66,32 @@ private void buildVertices(JorjeNode node, Vertex vNodeParam, String fileName) {
             vNode.property(Schema.FILE_NAME, fileName);
         }
 
-        Vertex vPreviousSibling = null;
         final List<JorjeNode> children = node.getChildren();
+        // Synthetic vertices create the possibility that children will be processed out of order.
+        // So instead of using an ArrayList, we use a HashMap whose keys are the indices.
+        final Map<Integer, Vertex> childVerticesByIndex = new HashMap<>();
         final Set<Vertex> verticesAddressed = new HashSet<>();
         verticesAddressed.add(vNode);
 
+        if (ConstructorUtil.isImplicitlyDelegatedConstructorBody(g, node)) {
+            // If this node is the body of a constructor that implicitly delegates to another
+            // constructor, synthesize vertices representing that delegation.
+            Vertex syntheticExpression = ConstructorUtil.synthesizeSuperCall(g, node, vNode);
+            verticesAddressed.add(syntheticExpression);
+            // Since the synthetic delegation is at the start, map its vertex as the 0th child
+            // and recompute the indices for the rest of the children.
+            childVerticesByIndex.put(0, syntheticExpression);
```

Comment:
please help me understand, why the index is hardcoded to a zero here?

Reply:
In this case, the synthesized vertex represents the `super()` call that's assumed to exist at the start of a subclass's constructor. So in order to make sure it exists at the start of the block, its index is 0.

Reply:
I see.  it would still help to read if the index was a constant that can describe what you described here